solidity/test/unit/XERC20.t.sol

```diff
@@ -30,20 +34,37 @@ contract UnitDeploy is Base {
   function testDeployment(
     uint256 _initialSupply
   ) public {
-    _xerc20 = new XERC20('Test', 'TST', _owner, _initialSupply, _receiver);
+    _xerc20 = new XERC20('Test', 'TST', _factory, _initialSupply, _receiver, _owner, _bridges, _limits, _limits);
     assertEq(XERC20(_xerc20).name(), 'Test');
     assertEq(XERC20(_xerc20).symbol(), 'TST');
     assertEq(XERC20(_xerc20).owner(), _owner);
+    assertEq(XERC20(_xerc20).FACTORY(), _factory);
     assertEq(XERC20(_xerc20).balanceOf(_receiver), _initialSupply);
   }
 
+  function testWithBridges(
+    uint256 _initialSupply
+  ) public {
+    _bridges.push(makeAddr('bridge1'));
+    _limits.push(1e18);
+
+    _xerc20 = new XERC20('Test', 'TST', _factory, _initialSupply, _receiver, _owner, _bridges, _limits, _limits);
+    assertEq(XERC20(_xerc20).name(), 'Test');
+    assertEq(XERC20(_xerc20).symbol(), 'TST');
+    assertEq(XERC20(_xerc20).owner(), _owner);
+    assertEq(XERC20(_xerc20).FACTORY(), _factory);
+    assertEq(XERC20(_xerc20).balanceOf(_receiver), _initialSupply);
+    assertEq(_xerc20.mintingMaxLimitOf(_bridges[0]), _limits[0]);
+    assertEq(_xerc20.burningMaxLimitOf(_bridges[0]), _limits[0]);
```

Comment:
Might be good to test with different limits for minting and burning.

Reply:
done